lib/AST/Type.cpp

```diff
@@ -4302,17 +4308,27 @@ case TypeKind::Id:
   case TypeKind::SILFunction: {
     auto fnTy = cast<SILFunctionType>(base);
     bool changed = false;
-
+    auto hasTypeErasedGenericClassType = [](Type ty) -> bool {
+      return ty.findIf([](Type subType) -> bool {
+        if (subType->getCanonicalType().isTypeErasedGenericClassType())
```

Comment:
getCanonicalType() should not be necessary here

Reply:
I'll fix on the master branch.

Reply:
Here: https://github.com/apple/swift/pull/33259